From 54fe18c6e64ddaf1de7ebf71733698eb8937c1c3 Mon Sep 17 00:00:00 2001
From: maurerr <mariusd84@gmail.com>
Date: Tue, 12 Feb 2019 15:15:59 +0100
Subject: [PATCH] no std for trunc and round

---
 i18n/decimfmt.cpp               | 2 +-
 i18n/number_decimalquantity.cpp | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/i18n/decimfmt.cpp b/i18n/decimfmt.cpp
index edd8910..6e67b4b 100644
--- a/i18n/decimfmt.cpp
+++ b/i18n/decimfmt.cpp
@@ -1349,7 +1349,7 @@ bool DecimalFormat::fastFormatDouble(double input, UnicodeString& output) const
         return false;
     }
     if (std::isnan(input)
-            || std::trunc(input) != input
+            || trunc(input) != input
             || input <= INT32_MIN
             || input > INT32_MAX) {
         return false;
diff --git a/i18n/number_decimalquantity.cpp b/i18n/number_decimalquantity.cpp
index 2c4182b..469252a 100644
--- a/i18n/number_decimalquantity.cpp
+++ b/i18n/number_decimalquantity.cpp
@@ -432,7 +432,7 @@ void DecimalQuantity::_setToDoubleFast(double n) {
         for (; i <= -22; i += 22) n /= 1e22;
         n /= DOUBLE_MULTIPLIERS[-i];
     }
-    auto result = static_cast<int64_t>(std::round(n));
+    auto result = static_cast<int64_t>(round(n));
     if (result != 0) {
         _setToLong(result);
         scale -= fracLength;
-- 
2.11.0

