From fcd4e85bf7d80400f06acaaf9f50d7ee1c00b701 Mon Sep 17 00:00:00 2001
From: maurerr <mariusd84@gmail.com>
Date: Fri, 8 Feb 2019 19:50:44 +0100
Subject: [PATCH] add BLKPBSZGET to diskio-unix.cc

---
 diskio-unix.cc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/diskio-unix.cc b/diskio-unix.cc
index d9f8b8d..f2eecfb 100644
--- a/diskio-unix.cc
+++ b/diskio-unix.cc
@@ -34,6 +34,7 @@
 #include <sstream>
 
 #include "diskio.h"
+#define BLKPBSZGET _IO(0x12,123)
 
 using namespace std;
 
-- 
2.11.0

