From ff122d191b6e1e3918d9643be7c755bdc91d9a76 Mon Sep 17 00:00:00 2001
From: maurerr <mariusd84@gmail.com>
Date: Sun, 10 Feb 2019 14:26:16 +0100
Subject: [PATCH] define missing I2C_SLAVE header

---
 server/drivers/hd44780-piplate.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/server/drivers/hd44780-piplate.c b/server/drivers/hd44780-piplate.c
index 170c147..4df9a27 100644
--- a/server/drivers/hd44780-piplate.c
+++ b/server/drivers/hd44780-piplate.c
@@ -64,6 +64,7 @@
 #endif
 
 #define DEFAULT_DEVICE		"/dev/i2c-1"
+#define I2C_SLAVE	0x0703	/* Use this slave address */
 
 void i2c_piplate_HD44780_senddata(PrivateData *p,
 				  unsigned char displayID, unsigned char flags, unsigned char ch);
-- 
2.11.0

