From 4892489f4aa5807d34f77e3a5809f9b1eb652ece Mon Sep 17 00:00:00 2001
From: maurerr <mariusd84@gmail.com>
Date: Sun, 10 Feb 2019 13:49:00 +0100
Subject: [PATCH] define missing pipe2 function

---
 redsocks.c | 19 +++++++++++++++++++
 1 file changed, 19 insertions(+)

diff --git a/redsocks.c b/redsocks.c
index f08cdbc..1045e8c 100644
--- a/redsocks.c
+++ b/redsocks.c
@@ -41,6 +41,25 @@
 
 #define REDSOCKS_RELAY_HALFBUFF  4096
 
+int __set_errno(int n)
+{
+	    errno = n;
+		    return -1;
+}
+
+int
+pipe2 (int pipedes[2], int flags)
+{
+	  if (pipedes == NULL)
+		      {
+				        __set_errno (EINVAL);
+						      return -1;
+							      }
+	    __set_errno (ENOSYS);
+		  return -1;
+}
+
+
 enum pump_state_t {
 	pump_active = -1,
 	pump_MAX = 0,
-- 
2.11.0

