From 92bb7c4df52e302e3cd845993f666b0999ea0158 Mon Sep 17 00:00:00 2001
From: maurerr <mariusd84@gmail.com>
Date: Sat, 26 Jan 2019 19:08:18 +0100
Subject: [PATCH] undefine gLibC

---
 src/unix/thread.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/unix/thread.c b/src/unix/thread.c
index 2900470..6f4effe 100644
--- a/src/unix/thread.c
+++ b/src/unix/thread.c
@@ -37,7 +37,7 @@
 #include <sys/sem.h>
 #endif
 
-#ifdef __GLIBC__
+#if !defined(__UCLIBC__)
 #include <gnu/libc-version.h>  /* gnu_get_libc_version() */
 #endif
 
-- 
2.11.0

