From 66e9c5e51041c4d256887203d2384756693a0db8 Mon Sep 17 00:00:00 2001
From: maurerr <mariusd84@gmail.com>
Date: Fri, 8 Feb 2019 16:30:00 +0100
Subject: [PATCH] fix libnfsidmap.c

---
 support/nfsidmap/libnfsidmap.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/support/nfsidmap/libnfsidmap.c b/support/nfsidmap/libnfsidmap.c
index 35ddf01..6688c08 100644
--- a/support/nfsidmap/libnfsidmap.c
+++ b/support/nfsidmap/libnfsidmap.c
@@ -61,6 +61,7 @@
 #include "nfsidmap_private.h"
 #include "nfsidmap_plugin.h"
 #include "conffile.h"
+#define NS_MAXMSG        65535        /*%< maximum message size */
 
 #pragma GCC visibility push(hidden)
 
-- 
2.11.0

