From 831acb4cb19aea98eccbfe916d49f12fd632b98a Mon Sep 17 00:00:00 2001
From: maurerr <mariusd84@gmail.com>
Date: Sat, 9 Feb 2019 15:04:24 +0100
Subject: [PATCH] added missing formats

---
 video_v4l2.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/video_v4l2.c b/video_v4l2.c
index 8b34127..c18122f 100644
--- a/video_v4l2.c
+++ b/video_v4l2.c
@@ -37,6 +37,9 @@
 #define MIN_MMAP_BUFFERS        2
 #define V4L2_PALETTE_COUNT_MAX 21
 
+#define V4L2_PIX_FMT_Y12     v4l2_fourcc('Y', '1', '2', ' ') /* 12  Greyscale     */
+#define V4L2_PIX_FMT_H264     v4l2_fourcc('H', '2', '6', '4') /* H264 with start codes */
+
 #define MAX2(x, y) ((x) > (y) ? (x) : (y))
 #define MIN2(x, y) ((x) < (y) ? (x) : (y))
 
-- 
2.11.0

