From 2192e4e2ca31dd753207a96424adc4ddfa1b3a57 Mon Sep 17 00:00:00 2001
From: maurerr <mariusd84@gmail.com>
Date: Wed, 6 Feb 2019 15:01:42 +0100
Subject: [PATCH] define O_CLOEXEC

---
 src/options.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/options.c b/src/options.c
index f5a7683..f83acfd 100644
--- a/src/options.c
+++ b/src/options.c
@@ -13,6 +13,10 @@
 #include "logging.h"
 #include "options.h"
 
+#ifndef O_CLOEXEC
+#define O_CLOEXEC        02000000        /* set close_on_exec */
+#endif
+
 void options_init(struct Options *opt) {
   opt->listen_addr = "127.0.0.1";
   opt->listen_port = 5053;
-- 
2.11.0

