From d61d34c52870e7b51e37368494d810d6559c7bcd Mon Sep 17 00:00:00 2001
From: maurerr <mariusd84@gmail.com>
Date: Mon, 11 Feb 2019 15:30:28 +0100
Subject: [PATCH] added pipe2 defintion to both gatt.c and gatt-client.c

---
 client/gatt.c     | 20 ++++++++++++++++++++
 src/gatt-client.c | 18 ++++++++++++++++++
 2 files changed, 38 insertions(+)

diff --git a/client/gatt.c b/client/gatt.c
index c7dfe42..63a7dee 100644
--- a/client/gatt.c
+++ b/client/gatt.c
@@ -43,6 +43,26 @@
 #include "gdbus/gdbus.h"
 #include "gatt.h"
 
+int __set_errno(int n)
+{
+           errno = n;
+           return -1;
+}
+
+int
+pipe2 (int pipedes[2], int flags)
+{
+        if (pipedes == NULL)
+        {
+        __set_errno (EINVAL);
+        return -1;
+        }
+           __set_errno (ENOSYS);
+           return -1;
+}
+	
+
+
 #define APP_PATH "/org/bluez/app"
 #define PROFILE_INTERFACE "org.bluez.GattProfile1"
 #define SERVICE_INTERFACE "org.bluez.GattService1"
diff --git a/src/gatt-client.c b/src/gatt-client.c
index 234f46e..36f704a 100644
--- a/src/gatt-client.c
+++ b/src/gatt-client.c
@@ -35,6 +35,24 @@
 
 #include "gdbus/gdbus.h"
 
+int __set_errno(int n)
+{
+   errno = n;
+   return -1;
+}
+
+int
+pipe2 (int pipedes[2], int flags)
+{
+    if (pipedes == NULL)
+            {
+            __set_errno (EINVAL);
+            return -1;
+            }
+    __set_errno (ENOSYS);
+    return -1;
+}
+
 #include "log.h"
 #include "error.h"
 #include "adapter.h"
-- 
2.11.0

