From a8c4f1affee02a99e379e69e0ce620816a961fe7 Mon Sep 17 00:00:00 2001
From: maurerr <mariusd84@gmail.com>
Date: Wed, 6 Feb 2019 12:35:02 +0100
Subject: [PATCH] added scsi_command_size_tbl function to sgio.c

---
 sgio.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/sgio.c b/sgio.c
index cb68418..7ccd957 100644
--- a/sgio.c
+++ b/sgio.c
@@ -17,6 +17,12 @@
 
 #include <linux/hdreg.h>
 
+const unsigned char scsi_command_size_tbl[8] =
+{
+		6, 10, 10, 12,
+			16, 12, 10, 10
+};
+
 extern int verbose;
 extern int prefer_ata12;
 
-- 
2.11.0

