From c0d7dde15532b111447161dda3c6c7ef2f743b80 Mon Sep 17 00:00:00 2001
From: maurerr <mariusd84@gmail.com>
Date: Thu, 7 Feb 2019 08:50:36 +0100
Subject: [PATCH] added malloc_usable_size structure

---
 lib/memory_mosq.c | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/lib/memory_mosq.c b/lib/memory_mosq.c
index 38e0c5d..ecb7a19 100644
--- a/lib/memory_mosq.c
+++ b/lib/memory_mosq.c
@@ -32,6 +32,19 @@ Contributors:
 #  endif
 #endif
 
+struct malloc_chunk {
+  size_t      prev_size;  /* Size of previous chunk (if free).  */
+  size_t      size;       /* Size in bytes, including overhead. */
+  struct malloc_chunk* fd;         /* double links -- used only if free. */
+  struct malloc_chunk* bk;
+};
+#define OVERHEAD (2*sizeof(size_t))
+#define CHUNK_SIZE(c) ((c)->size & -2)
+#define MEM_TO_CHUNK(p) (struct malloc_chunk *)((char *)(p) - OVERHEAD)
+size_t malloc_usable_size(void *p) {
+	return p ? CHUNK_SIZE(MEM_TO_CHUNK(p)) - OVERHEAD : 0;
+}
+
 #ifdef REAL_WITH_MEMORY_TRACKING
 static unsigned long memcount = 0;
 static unsigned long max_memcount = 0;
-- 
2.11.0

