From f7cb15ea29263b5bf391da19fd4b15c11fd693d7 Mon Sep 17 00:00:00 2001
From: maurerr <mariusd84@gmail.com>
Date: Fri, 8 Feb 2019 09:50:20 +0100
Subject: [PATCH] remove __builtin_choose_expr

---
 erts/emulator/beam/sys.h | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/erts/emulator/beam/sys.h b/erts/emulator/beam/sys.h
index bb22548..135eef9 100644
--- a/erts/emulator/beam/sys.h
+++ b/erts/emulator/beam/sys.h
@@ -232,17 +232,10 @@ __decl_noreturn void __noreturn erl_assert_error(const char* expr, const char *f
  * Compile time assert
  * (the actual compiler error msg can be a bit confusing)
  */
-#if ERTS_AT_LEAST_GCC_VSN__(3,1,1)
-# define ERTS_CT_ASSERT(e) \
-    do { \
-	enum { compile_time_assert__ = __builtin_choose_expr((e),0,(void)0) }; \
-    } while(0)
-#else
 # define ERTS_CT_ASSERT(e) \
     do { \
         enum { compile_time_assert__ = 1/(e) }; \
     } while (0)
-#endif
 
 /*
  * Microsoft C/C++: We certainly want to use stdarg.h and prototypes.
-- 
2.11.0

