From b8cc3bb1776ca7a2f5b9d4473f5c085bd793813b Mon Sep 17 00:00:00 2001
From: maurerr <mariusd84@gmail.com>
Date: Sat, 9 Feb 2019 21:39:13 +0100
Subject: [PATCH] added O_CLOEXEC definition

---
 src/fs.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/fs.c b/src/fs.c
index d0e3f7e..b5a75ab 100644
--- a/src/fs.c
+++ b/src/fs.c
@@ -1,6 +1,7 @@
 #include "common.h"
 #include "id.h"
 #include "fs.h"
+#define O_CLOEXEC      02000000
 
 int direxists(const char *dir)
 {
-- 
2.11.0

